app/model/parcel.py

```diff
@@ -0,0 +1,247 @@
+            response.headers['Location'] = "parcels/" + str(parcel['id'])
+            return response
+        else:
+            response = Response(json.dumps({"error": "Invalid Parcel delivery order object"}),
+                                status=400, mimetype="application/json")
+            return response
+
+    def get_all_parcel(self):
+        """
+        get all parcels posted
+        """
+        return self.parcels
+
+    def get_one_parcel(self, id):
+        """
+        get one parcel by its id
+        """
+        for parcel in self.parcels:
+            if parcel['id'] == id:
+                return parcel
+
+    def cancel_parcel(self, id):
+        global cancelled_parcel
+        for parcel in self.parcels:
+            if parcel['id'] == id:
+                cancelled_parcel = {
+                    'id': parcel['id'],
+                    'pickup_address': parcel['pickup_address'],
+                    'destination_address': parcel['destination_address'],
+                    'comment_description': parcel['comment_description'],
+                    'status': "cancelled",
+                    'current_location': parcel['current_location'],
+                    'created': parcel['created'],
+                    'user_id': parcel['user_id'],
+                    'recipient_address': parcel['recipient_address'],
+                    'recipient_phone': parcel['recipient_phone'],
+                    'recipient_email': parcel['recipient_email'],
+                    'weight': parcel['weight'],
+                    'distance': self.get_distance(parcel['pickup_address'], parcel['destination_address']),
+                    'pick_up_lat_lng': self.getpickuplatlng(parcel['pickup_address']),
+                    'destination_lat_lng': self.getdestinationlatlng(parcel['destination_address']),
+                    'price': self.get_charge(parcel['weight'],
+                                             self.get_distance(parcel['pickup_address'], parcel['destination_address']))
+
+                }
+                parcel.update(cancelled_parcel)
+
+        return cancelled_parcel
+
+    def delete_one_parcel(self, id):
+        """ delete an parcel by its id
+        """
+        parcel = self.get_one_parcel(id)
+        if parcel:
+            self.parcels.remove(parcel)
+            return "deleted successfully"
+
+    def is_valid_request(self, newparcel):
+        if "destination_address" in newparcel and "pickup_address" in newparcel \
+                and "comment_description" in newparcel and "created" in newparcel and \
+                "user_id" in newparcel and "recipient_address" in newparcel and "recipient_phone" in newparcel and \
+                "recipient_email" in newparcel and "status" in newparcel and "recipient_name" in newparcel and "weight" in newparcel:
+            return True
+        else:
+            return False
+
+    def is_order_delivered(self, id):
+        '''
```

Comment:
Maintain a uniform way of documenting your work, if you are using single quotes `'''`, maintain that, if you are using double quotes `"""`, stick to that